backend/app/routes/web/recommendation.py

```diff
@@ -27,58 +30,82 @@ def _get_panel(panel_id: str) -> list[str]:
         return ["software"]
     elif panel_id == "training":
         return ["trainings"]
-    elif panel_id == "service":
-        # IMPORTANT!!! recommender does not support services
-        return []
-
-
-async def _get_recommendations(client: AsyncClient, session: SessionData, panel_id: str):
-    panel_id_options = ["publication", "datasets", "software", "training"]
-    page_id = "/search/" + panel_id
-    panels = [random.choice(panel_id_options)] if panel_id == "all" else _get_panel(panel_id)
-    request_body = {
-        "user_id": session.aai_id,
-        "unique_id": session.session_uuid,
-        "timestamp": datetime.datetime.utcnow().isoformat()[:-3] + "Z",
-        "visit_id": str(uuid.uuid4()),
-        "page_id": page_id,
-        "panel_id": panels,
-        "candidates": [],
-        "search_data": {},
-    }
-
-    response = await client.post(
-        RECOMMENDER_ENDPOINT,
-        json=request_body,
-    )
 
-    if response.status_code != 200:
-        return None
+    return []
 
-    recommendation_data = response.json()
-    if len(recommendation_data) != 1:
-        return None
 
-    return recommendation_data
+async def _get_recommended_uuids(
+    client: AsyncClient, session: SessionData, panel_id: str
+):
+    try:
+        panel_id_options = ["publication", "datasets", "software", "training"]
+        page_id = "/search/" + panel_id
+        panels = (
+            [random.choice(panel_id_options)]
+            if panel_id == "all"
+            else _get_panel(panel_id)
+        )
+        request_body = {
+            "user_id": session.aai_id,
+            "unique_id": session.session_uuid,
+            "timestamp": datetime.datetime.utcnow().isoformat()[:-3] + "Z",
+            "visit_id": str(uuid.uuid4()),
+            "page_id": page_id,
+            "panel_id": panels,
+            "candidates": [],
+            "search_data": {},
+        }
+
+        response = await client.post(
+            RECOMMENDER_ENDPOINT,
+            json=request_body,
+        )
+
+        if response.status_code != 200:
+            return await _get_random_ids(client)
+
+        recommendation_data = response.json()
+        if len(recommendation_data) != 1:
+            return await _get_random_ids(client)
+
+        return recommendation_data[0]["recommendations"]
+    except httpx.ConnectError:
+        return await _get_random_ids(client)
+
+
+async def _get_random_ids(client: AsyncClient):
+    try:
+        facets = {
+            "id": TermsFacet(field="id", type="terms", limit=100),
+        }
+        response = await search(
+            client=client,
+            collection="all_collection",
+            q="*",
```

Comment:
Filter parameters should be taken from the recommendations request context.
Furthermore, it will cause a bug in which item space (from which we can serve recommendations) is empty or has only 1-2 resources, we will serve 3 recommendations from the whole global item space